### PR TITLE
fix: correct file path in snapshot checksum and add line terminater

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:

-

## Reference issue to close (if applicable)

Closes
https://github.com/ChainSafe/forest/issues/2835
https://github.com/ChainSafe/forest/issues/2836

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: node/rpc/src/chain_api.rs [hunk 1]
```diff
@@ -150,7 +150,16 @@ where
 /// Prints hex-encoded representation of SHA-256 checksum and saves it to a file
 /// with the same name but with a `.sha256sum` extension.
 async fn save_checksum(source: &Path, hash: Output<Sha256>) -> Result<()> {
-    let encoded_hash = format!("{} {}", hash.encode_hex::<String>(), source.display());
+    let encoded_hash = format!(
+        "{} {}\n",
+        hash.encode_hex::<String>(),
+        source
+            .file_name()
+            .and_then(|i| i.to_str())
```
Comment (LesnyRumcajs) [inline node/rpc/src/chain_api.rs:158]:
```suggestion
            .and_then(|name| name.to_str())
```

File: node/rpc/src/chain_api.rs [hunk 2]
```diff
@@ -150,7 +150,16 @@ where
...
+            .ok_or_else(|| anyhow::Error::msg(
```
Comment (LesnyRumcajs) [inline node/rpc/src/chain_api.rs:159]: will `anyhow!` macro work here?

**Review by LesnyRumcajs — APPROVED** [review]

**Review by lemmih — CHANGES_REQUESTED** [review]
You're logging the `encoded_hash` which contains a newline. You probably meant to log the hash string.

File: node/rpc/src/chain_api.rs (hunk 2 above)
Comment (aatifsyed) [reply node/rpc/src/chain_api.rs:159]: I'd use `.context("Failed to retrieve file name while saving checksum")`

File: node/rpc/src/chain_api.rs (hunk 1 above)
Comment (aatifsyed) [reply node/rpc/src/chain_api.rs]: `OsStr::to_str`, no need to make a closure

File: node/rpc/src/chain_api.rs (hunk 2 above)
Comment (hanabi1224) [reply node/rpc/src/chain_api.rs:159]: @aatifsyed The output of `and_then` is `Option`, I'm not sure if I can use `.context` to convert it to error

Comment (lemmih) [reply node/rpc/src/chain_api.rs:159]: @hanabi1224 `.context` works for both `Result` and `Option`.

Comment (hanabi1224) [reply node/rpc/src/chain_api.rs]: Fixed

**Comment by hanabi1224** [comment]
> You're logging the `encoded_hash` which contains a newline. You probably meant to log the hash string.

Fixed

File: node/rpc/src/chain_api.rs [hunk 3]
```diff
@@ -150,13 +150,22 @@ where
 /// Prints hex-encoded representation of SHA-256 checksum and saves it to a file
 /// with the same name but with a `.sha256sum` extension.
 async fn save_checksum(source: &Path, hash: Output<Sha256>) -> Result<()> {
-    let encoded_hash = format!("{} {}", hash.encode_hex::<String>(), source.display());
+    let encoded_hash = hash.encode_hex::<String>();
+    let checksum_file_content = format!(
+        "{encoded_hash} {}\n",
+        source
+            .file_name()
+            .and_then(|name| name.to_str())
```
Comment (lemmih) [inline node/rpc/src/chain_api.rs:158]: I quite liked Aatif's suggestion:
```suggestion
            .and_then(std::ffi::OsStr::to_str)
```
:)

**Review by lemmih — APPROVED** [review]

**Pull request merged**